shiny/ui/_chat.py

```diff
@@ -585,7 +585,9 @@ async def _append_message_stream(self, message: AsyncIterable[Any]):
         id = _utils.private_random_id()
 
         empty = ChatMessage(content="", role="assistant")
-        await self._append_message(empty, chunk="start", stream_id=id)
+        async for msg in message:
+            await self._append_message(msg, chunk="start", stream_id=id)
+            break
```

Comment:
This changes mean that the stream is now iterated over (and sent) *twice*. Is that intentional?

Reply:
The first for loop waits for the first message and then breaks. The next for loop sees all the rest of the messages.

```python
>>> "hello".__iter__()
<str_iterator object at 0x104882920>
>>> it = "hello".__iter__()
>>> for c in it: print(c)
...
h
e
l
l
o
>>> for c in it: print(c)
...
>>>
```

Although now looking at this again, this is probably much more confusing looking than it needs to be, rather than just having a single loop that keeps a flag of whether it's sent any values yet or not.

Reply:
Ah, sorry, misread the intention. 

> a single loop that keeps a flag of whether it's sent any values yet or not.

That or perhaps use `next()`?